source/hic_hal/stm32/stm32h743ii/adc.c

```diff
@@ -0,0 +1,105 @@
+#include "adc.h"
+#include "stm32h743xx.h"
+#include "stm32h7xx_hal.h"
+
+// Timeout value in millisecond
+#define ADC_CONVERSION_TIMEOUT (10)
+#define ADC_BASE ADC3
+
+static ADC_HandleTypeDef s_adc_handle;
+static ADC_ChannelConfTypeDef s_chan_conf;
+
+static void error_handler(void)
+{
+    while (1)
+    {
+    }
+}
+
+void adc_init(void)
+{
+    s_adc_handle.Instance = ADC_BASE;
+    if (HAL_ADC_DeInit(&s_adc_handle) != HAL_OK)
+    {
+        error_handler();
+    }
+    s_adc_handle.Init.ClockPrescaler = ADC_CLOCK_ASYNC_DIV2;
+    s_adc_handle.Init.Resolution = ADC_RESOLUTION_16B;
+    s_adc_handle.Init.ScanConvMode = DISABLE;
+    s_adc_handle.Init.EOCSelection = ADC_EOC_SINGLE_CONV;
+    s_adc_handle.Init.LowPowerAutoWait = DISABLE;
+    s_adc_handle.Init.ContinuousConvMode = DISABLE;
+    s_adc_handle.Init.NbrOfConversion = 1;
+    s_adc_handle.Init.DiscontinuousConvMode = DISABLE;
+    s_adc_handle.Init.NbrOfDiscConversion = 1;
+    s_adc_handle.Init.ExternalTrigConv = ADC_SOFTWARE_START;
+    s_adc_handle.Init.ExternalTrigConvEdge = ADC_EXTERNALTRIGCONVEDGE_NONE;
+    s_adc_handle.Init.ConversionDataManagement = ADC_CONVERSIONDATA_DR;
+    s_adc_handle.Init.Overrun = ADC_OVR_DATA_OVERWRITTEN;
+    s_adc_handle.Init.OversamplingMode = DISABLE;
+
+    if (HAL_ADC_Init(&s_adc_handle) != HAL_OK)
+    {
+        error_handler();
+    }
+}
+
+void adc_init_pins(void)
+{
+}
+
+uint32_t adc_read_channel(uint32_t channelGroup, uint32_t channelNumber, uint32_t channelMux)
+{
+    uint16_t udc_adapter_adc_value;
```

Comment:
I think just adc_value is ok. This looks like it's supposed to be somewhat generic, so not limited to reading the UDC adapter type, is that right?